### PR TITLE
match baldr pathlocation refactorings

### PR DESCRIPTION
**Pull request by kevinkreiser**
baldr pathlocation got a refactor in this pr: valhalla/baldr#254

the main reason for this was to allow a more sensible way for loki to correlate more than one result per input. this is to help solve the problem of islands of connectivity valhalla/loki#127

ive also open up an issue here #24 because i think espesially in map_matching.cc there is a lot of loki-like work being done and just having meili make use of loki directly might be a good thing. see the issue for more details.

### CODE REVIEW

**Comment by kevinkreiser** [comment]
id like to merge the baldr pr so this pr would need to be merged as well. would anyone mind giving me a ship?

File: src/meili/map_matching.cc [hunk 1]
```diff
@@ -413,8 +396,10 @@ interpolate(baldr::GraphReader& reader,
 
   for (auto candidate = begin; candidate != end; candidate++) {
     if (candidate->sq_distance() < closest_sq_distance) {
-      if (!candidate->IsNode()) {
-        for (const auto& edge : candidate->edges()) {
+      //Note: technically a candidate can have correlated to more than one place in the graph
+      //but the way its used in meili we only correlated it to one place so .front() is safe
+      if (!candidate->edges.front().begin_node() && !!candidate->edges.front().end_node()) {
```
Comment (ptpt) [inline src/meili/map_matching.cc]: double !!?

**Comment by ptpt** [comment]
:ship: when the double negative fixed

File: src/meili/map_matching.cc (hunk 1 above)
Comment (kevinkreiser) [inline src/meili/map_matching.cc]: excellent catch! sorry about that

**Pull request merged**